### PR TITLE
Switch to new workflows

### PR DESCRIPTION
**Pull request by uncleDecart**
Following work done to split eden test workflows in https://github.com/lf-edge/eden/issues/863 and integration
of this workflows in EVE repository, this commit switches to use new
workflows in eden itself

Also, new workflows are running on Buildjet runners, however, they
are not available in personal forks and it is useful to run test.yml
in fork, that is why this commit introduces determine-runner job,
which checks if we are in fork and changes runner to ubuntu-2204 which
is available everywhere

### CODE REVIEW

**Comment by uncleDecart** [comment]
I did not test it yet in my fork and would be great if @yash-zededa can check workflow files as well

File: .github/workflows/test.yml
Comment (milan-zededa) [inline .github/workflows/test.yml:23]: Would it be possible to extend this check down to `action/setup-environment` and enable `eve.accel` if workflow is running on buildjet?

Comment (uncleDecart) [reply .github/workflows/test.yml:23]: I thought we could actually change setup-environment and use some utilities to determine if hw acceleration is possible, that way we don't have coupling with runner, but rather with capability

File: .github/workflows/eden.yml
Comment (giggsoff) [inline .github/workflows/eden.yml:22]: Should we split the PR into two if you would like to modify the file you point onto here, but it doesn't exists in master branch?

File: .github/workflows/eden.yml
Comment (giggsoff) [inline .github/workflows/eden.yml:11]: We had run without approval before, well, it is debatable and probably not good for some kind of paid runners. But is it possible to implement workflow_dispatch for users with permissions to not wait for approval...

File: .github/workflows/eden.yml
Comment (uncleDecart) [reply .github/workflows/eden.yml:22]: this file exists in master, only change here is runners for forks will be ubuntu :D

File: .github/workflows/eden.yml
Comment (uncleDecart) [reply .github/workflows/eden.yml:11]: We can actually restore previous permissions, but is it taking long time to wait for approval in eden repository?

Comment (uncleDecart) [reply .github/workflows/eden.yml]: moved it back to the way it was

**Comment by uncleDecart** [comment]
Something weird is happening with GHA I see error which I shouldn't see [here](https://github.com/lf-edge/eden/actions/runs/6587404317) will retry in some time

File: .github/workflows/eden.yml
Comment (uncleDecart) [reply .github/workflows/eden.yml]: syntax is GHA is tricky, my intention was to use local action, fixed to use just that. GHA seems to work, I'll wait for it to finish.

Comment (giggsoff) [reply .github/workflows/eden.yml]: [It](https://github.com/lf-edge/eden/actions/runs/6587792704/job/17898759398?pr=909#step:2:9) looks like another problem. It is really tricky

File: .github/workflows/test.yml
Comment (giggsoff) [inline .github/workflows/test.yml:26]: It was `ubuntu-22.04` before. Also please solve https://github.com/lf-edge/eden/pull/909#discussion_r1366923094

File: .github/workflows/eden.yml
Comment (uncleDecart) [reply .github/workflows/eden.yml]: That's just my poor bash-ninja skills :D

Comment (giggsoff) [reply .github/workflows/eden.yml]: Probably 
```
[[ "${{ github.event.repository.full_name}}" = "lf-edge/eve" ]] || [[ "${{ github.event.repository.full_name}}" = "lf-edge/eden" ]]
```

Comment (uncleDecart) [reply .github/workflows/eden.yml]: Seems to be [working](https://github.com/lf-edge/eden/actions/runs/6588817373/job/17902163780)

Comment (uncleDecart) [reply .github/workflows/eden.yml]: I think it'll make sense to include similar thing to eden_setup to speed up process in eden repository

**Comment by uncleDecart** [comment]
onborading seems to [fail](https://github.com/lf-edge/eden/actions/runs/6588817373/job/17902163780). Checked locally with master, works

**Comment by giggsoff** [comment]
I can see `time="2023-10-20T14:36:39Z" level=fatal msg="Setup eden failed: cannot setup EVE: tag not present"`. So it looks like a problem with the check [here](https://github.com/lf-edge/eden/blob/06035a03038f81900ce3f1a68b60f9e4c025da50/.github/actions/setup-environment/action.yml#L53). But it looks solid...

**Comment by uncleDecart** [comment]
I think you're right @giggsoff, let me check your changes

**Comment by uncleDecart** [comment]
Trying to make it work https://media.giphy.com/media/OVwfSuQXVUUI8/giphy.gif

**Comment by uncleDecart** [comment]
Now it's working, but we need to check if it works with EVE

**Comment by uncleDecart** [comment]
I think that we can merge it, try to run manually with eve_image on master branch and if it doesn't work, we do more fixes before, releasing eden and switching on EVE. Any objections @giggsoff @yash-zededa ?

**Comment by giggsoff** [comment]
Only one question here: as I understand, we expect failure of tests because of https://github.com/lf-edge/eden/issues/902. Will we fix the problem before experiments in EVE? Here as Milan suggested or separately?

File: .github/workflows/test.yml
Comment (milan-zededa) [reply .github/workflows/test.yml]: Something like this: https://manpages.ubuntu.com/manpages/impish/man1/kvm-ok.1.html ?

Comment (uncleDecart) [reply .github/workflows/test.yml]: Yes, I also found option with lscpu :), sending it rn

File: .github/actions/setup-environment/action.yml
Comment (giggsoff) [inline .github/actions/setup-environment/action.yml:47]: Can you please check line with e.g. https://www.shellcheck.net?;)

Comment (uncleDecart) [reply .github/actions/setup-environment/action.yml]: This PR should be renamed as "My attempt to gain 3rd dan in bash" :D

**Comment by uncleDecart** [comment]
Feels like it [hanged](https://github.com/lf-edge/eden/actions/runs/6615955772/job/17969404064?pr=909) on onboarding, any ideas @giggsoff ?

**Comment by giggsoff** [comment]
> Feels like it [hanged](https://github.com/lf-edge/eden/actions/runs/6615955772/job/17969404064?pr=909) on onboarding, any ideas @giggsoff ?

Let's wait for logs from console

File: .github/actions/setup-environment/action.yml
Comment (giggsoff) [inline .github/actions/setup-environment/action.yml:59]: Curious, what is `input_string` here? From where it comes?

Comment (uncleDecart) [reply .github/actions/setup-environment/action.yml:59]: That's from my bash ninja malpractices. I'll fix it tommorow. In should've been image

**Comment by uncleDecart** [comment]
@giggsoff  there's something going on with eden tests and connecting to controller (https://github.com/lf-edge/eve/actions/runs/6612672662/job/17958950820?pr=3514)

**Comment by milan-zededa** [comment]
> @giggsoff there's something going on with eden tests and connecting to controller (https://github.com/lf-edge/eve/actions/runs/6612672662/job/17958950820?pr=3514)

```
time="2023-10-23T11:39:04Z" level=fatal msg="Setup eden failed: cannot setup EVE: cannot download EVE: ImagePull (evebuild/danger:pr3514-kvm-amd64): imagePull: Error response from daemon: manifest for evebuild/danger:pr3514-kvm-amd64 not found: manifest unknown: manifest unknown"
```

**Comment by uncleDecart** [comment]
>> @giggsoff there's something going on with eden tests and connecting to controller (https://github.com/lf-edge/eve/actions/runs/6612672662/job/17958950820?pr=3514)

> time="2023-10-23T11:39:04Z" level=fatal msg="Setup eden failed: cannot setup EVE: cannot download EVE: ImagePull (evebuild/danger:pr3514-kvm-amd64): imagePull: Error response from daemon: manifest for evebuild/danger:pr3514-kvm-amd64 not found: manifest unknown: manifest unknown"

Maybe it makes sense to condition EVE workflows on published build...

**Comment by uncleDecart** [comment]
@giggsoff @milan-zededa I've hardcoded EVE version to test for 10.11.0 which is same to default EVE version, because it was failing with no tag as. Not sure why, because there's only one ./eden config set default --key=eve.tag --value, but It's not invoked. I believe that we can leave it for now, since priority is to write first golang test.
But I had question on why do we do build and setup if we do same thing in test?

File: .github/workflows/test.yml
Comment (giggsoff) [reply .github/workflows/test.yml]: I am not sure that current logic works. I can see SandyBridge [here](https://github.com/lf-edge/eden/actions/runs/6642450932/job/18047461117?pr=909#step:3:1476), but it should be [host](https://github.com/lf-edge/eden/blob/a0dbf14c57ec8f9c7897ea9bbb569a800bc23462/pkg/defaults/defaults.go#L126) in case of acceleration enabled.

**Comment by giggsoff** [comment]
> why do we do build and setup if we do same thing in test?

We should build to run tests as they are binaries (sorry if I did not understand question here). Setup is inside test, correct, but we want to do some specific preparation for zfs case with specific options… So we detached the logic into the workflow itself and just skip setup in test if already done. Potentially we can prepare some arguments/envs for tests, but that way we will mostly hide the logic without any improvements.

**Comment by uncleDecart** [comment]
So https://github.com/lf-edge/eden/actions/runs/6642450932/job/18050451195?pr=909#step:3:5450 failed again, but it didn't fetch latest workflow file because I assume it didn't restart all jobs but failed. Interesting doc [link](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs#re-running-workflows-and-jobs-with-reusable-workflows)

**Comment by giggsoff** [comment]
> So VNC test [failed](https://github.com/lf-edge/eden/actions/runs/6642450932/job/18050451195?pr=909#step:3:5450) again, but it didn't fetch latest workflow file because I assume it didn't restart all jobs but failed. Interesting doc [link](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs#re-running-workflows-and-jobs-with-reusable-workflows)

Interesting link, you are right. 
It is not clear what is re-try. Will it applies to re-run truggered by re-push? Or only manual re-try triggered by re-try button. 
Also it is not clear, will it applies only for workflow files itself or also to files in repo. For example, if I modify/implement test, can I check it inside PR or it is not possible with reusable workflow and we should merge changes before check?

**Comment by uncleDecart** [comment]
>>So VNC test [failed](https://github.com/lf-edge/eden/actions/runs/6642450932/job/18050451195?pr=909#step:3:5450) again, but it didn't fetch latest workflow file because I assume it didn't restart all jobs but failed. Interesting doc [link](https://docs.github.com/en/actions/managing-workflow-runs/re-running-workflows-and-jobs#re-running-workflows-and-jobs-with-reusable-workflows)

>Interesting link, you are right.
It is not clear what is re-try. Will it applies to re-run truggered by re-push? Or only manual re-try triggered by re-try button.
Also it is not clear, will it applies only for workflow files itself or also to files in repo. For example, if I modify/implement test, can I check it inside PR or it is not possible with reusable workflow and we should merge changes before check?

From what I understand it means manually pressing re-run all jobs button, but stay tuned. We are trying it right now :D

**Comment by giggsoff** [comment]
> Okay, so I don't see not based on OCI image, which indicates that VNC is fixed and with this PR #902 will be resolved. Runners are correctly used, we need to check only that EVE is pulling correct version on tag (we can do that after releasing 0.9.3)

1. I still can see SandyBridge [here](https://github.com/lf-edge/eden/actions/runs/6642450932/job/18125731904?pr=909#step:3:1454), which indicates that we do not use kvm (otherwise it will be host cpu in the line). Probably we just end the tests forkflow before the VNC test?
2. It will not solve the issue you point onto for non-accelerated runners, see the comment https://github.com/lf-edge/eden/issues/902#issuecomment-1779613660

**Comment by giggsoff** [comment]
Really strange, you modified the test, but new lines around [that](https://github.com/lf-edge/eden/actions/runs/6767802891/job/18404850742?pr=909#step:3:5361) one are not come into the workflow…
Maybe the problem is somewhere in checkout step? Looks like it [try](https://github.com/lf-edge/eden/actions/runs/6767802891/job/18404850742?pr=909#step:2:481) to pull hash from master branch.

**Comment by uncleDecart** [comment]
So it gets curiouser and curiouser. I found this 

https://github.com/actions/checkout/issues/1418

Let's try to checkout to 3.5.3

File: .github/workflows/test.yml
Comment (giggsoff) [inline .github/workflows/test.yml:81]: Looks like a duplicate

File: .github/actions/setup-environment/action.yml
Comment (giggsoff) [inline .github/actions/setup-environment/action.yml:37]: Seems `lscpu` comes from `util-linux` package.

**Comment by giggsoff** [comment]
Please rebase on top of master with https://github.com/lf-edge/eden/pull/924 merged

**Comment by giggsoff** [comment]
Well, I can see host cpu in EVE-OS start [line](https://github.com/lf-edge/eden/actions/runs/6798825371/job/18485068406#step:3:1490). So we use acceleration. Let's wait for the rest of the test. But seems we should workaround cloud-init test (revert changes or stabilize and adjust EVE-OS hash).

**Comment by giggsoff** [comment]
New day, new problems: I can see _The self-hosted runner lost communication with the server_  again and again https://github.com/lf-edge/eden/actions/runs/6811998752?pr=909

**Comment by uncleDecart** [comment]
I'll ask BuildJet about it

**Review by giggsoff — APPROVED** [review]

**Review by milan-zededa — APPROVED** [review]

**Pull request merged**
